File: src/scripts/seed.py
"""种子脚本：创建默认管理员账号
运行方式: python3 src/scripts/seed.py
默认管理员: admin@worldcup.com / admin123

使用 hashlib + scrypt 模拟 bcrypt 哈希（生产环境会使用 bcryptjs 从 Next.js 端处理）
"""

import sqlite3
import hashlib
import uuid
import binascii
from datetime import datetime, timezone

DB_PATH = "prisma/dev.db"
ADMIN_EMAIL = "admin@worldcup.com"
ADMIN_PASSWORD = "admin123"

def hash_password(password: str) -> str:
    """使用 Python hashlib 生成与 bcrypt 兼容的哈希。
    由于种子数据仅在开发环境使用，这里使用 pbkdf2 + base64 模拟。
    注意: 实际登录时 Auth.js 会使用 bcryptjs 验证。
    """
    salt = b"$2b$12$0000000000000000000000"  # dummy salt
    dk = hashlib.pbkdf2_hmac('sha256', password.encode(), salt, 100000)
    # 返回 bcrypt 风格的占位哈希，实际值由 bcryptjs 生成
    return f"$2b$12${binascii.hexlify(dk).decode()[:53]}"

def main():
    conn = sqlite3.connect(DB_PATH)
    conn.row_factory = sqlite3.Row
    cur = conn.cursor()

    existing = cur.execute("SELECT * FROM User WHERE email = ?", (ADMIN_EMAIL,)).fetchone()

    if existing:
        print(f"✅ 管理员账号已存在: {ADMIN_EMAIL}")
        if existing["role"] != "admin":
            cur.execute("UPDATE User SET role = ? WHERE email = ?", ("admin", ADMIN_EMAIL))
            conn.commit()
            print("✅ 已更新为管理员角色")
    else:
        # 先用占位哈希，实际密码验证由 bcryptjs 处理
        hashed = hash_password(ADMIN_PASSWORD)
        user_id = str(uuid.uuid4())
        now = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.000Z")

        cur.execute(
            "INSERT INTO User (id, email, password, name, role, createdAt, updatedAt) VALUES (?, ?, ?, ?, ?, ?, ?)",
            (user_id, ADMIN_EMAIL, hashed, "管理员", "admin", now, now)
        )
        conn.commit()
        print(f"✅ 管理员账号已创建 (占位哈希):")
        print(f"   邮箱: {ADMIN_EMAIL}")
        print(f"   密码: {ADMIN_PASSWORD}")
        print(f"   ⚠️  哈希为占位值，请在 Next.js 中通过注册接口重新设置密码")

    conn.close()
    print("\n🎉 种子数据准备完成！")

if __name__ == "__main__":
    main()

File: src/scripts/seed.mjs
// 种子脚本：创建默认管理员账号
// 运行方式: node src/scripts/seed.mjs
// 默认管理员: admin@worldcup.com / admin123

import { neon } from "@neondatabase/serverless";
import bcrypt from "bcryptjs";
import { randomUUID } from "node:crypto";

const DATABASE_URL = process.env.DATABASE_URL;
if (!DATABASE_URL) {
  console.error("❌ 请设置 DATABASE_URL 环境变量");
  process.exit(1);
}

const sql = neon(DATABASE_URL);

const adminEmail = "admin@worldcup.com";
const adminPassword = "admin123";

async function main() {
  // Check if admin exists
  const rows = await sql`SELECT * FROM "User" WHERE email = ${adminEmail}`;

  if (rows.length > 0) {
    console.log(`✅ 管理员账号已存在: ${adminEmail}`);
    if (rows[0].role !== "admin") {
      await sql`UPDATE "User" SET role = 'admin' WHERE email = ${adminEmail}`;
      console.log("✅ 已更新为管理员角色");
    }
  } else {
    const hashedPassword = bcrypt.hashSync(adminPassword, 12);
    const id = randomUUID();
    const now = new Date().toISOString();

    await sql`
      INSERT INTO "User" (id, email, password, name, role, "createdAt", "updatedAt")
      VALUES (${id}, ${adminEmail}, ${hashedPassword}, ${'管理员'}, ${'admin'}, ${now}, ${now})
    `;

    console.log(`✅ 管理员账号已创建:`);
    console.log(`   邮箱: ${adminEmail}`);
    console.log(`   密码: ${adminPassword}`);
  }

  console.log("\n🎉 种子数据准备完成！");
  process.exit(0);
}

main().catch((e) => {
  console.error("❌ 种子脚本执行失败:", e);
  process.exit(1);
});
